src/Symfony/Component/Security/Core/User/LdapUserProvider.php

```diff
@@ -113,30 +118,31 @@ public function supportsClass($class)
      */
     protected function loadUser($username, Entry $entry)
     {
-        $password = $this->getPassword($entry);
+        $password = $this->getAttributeValue($entry, $this->passwordAttribute);
 
         return new User($username, $password, $this->defaultRoles);
     }
 
     /**
-     * Fetches the password from an LDAP entry.
+     * Fetches a required unique attribute value from an LDAP entry.
      *
      * @param null|Entry $entry
+     * @param string     $attribute
      */
-    private function getPassword(Entry $entry)
+    private function getAttributeValue(Entry $entry, $attribute)
     {
-        if (null === $this->passwordAttribute) {
+        if (null === $attribute) {
```

Comment:
I'm not really sure to understand what you expect here.

About moving the logic inside `loadUser()` (moving the `getAttributeValue()` at line 86 to the loadUser method), see my previous comment. It is what I wanted to do initially but it would introduce depreciations or BC breaks.

If we want to allow an empty password, this would change the initial behaviour and I think it would be better to introduce the change in another PR.

But I agree, password attribute should not be mandatory here.